services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaController.kt

```diff
@@ -434,6 +438,78 @@ Returns the updated hankekayttaja.
         return hankeKayttajaService.updateKayttajaInfo(hankeTunnus, update, kayttajaId).toDto()
     }
 
+    @GetMapping("/kayttajat/{kayttajaId}/deleteInfo")
+    @Operation(
+        summary = "Check if user can be deleted",
+        description =
+            """
+Check if there are active applications the user is a contact in that would
+prevent their deletion. Deleting would also be prevented if the user is the only
+contact in the applicant customer role, so check that as well.
+
+Also, collect draft applications the user is a contact in, since the frontend
+needs to show them when asking for confirmation.
+
+Does not check if the caller needs KAIKKI_OIKEUDET to delete this particular
+user.
+"""
+    )
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(
+                    description = "Return the results",
+                    responseCode = "200",
+                ),
+                ApiResponse(
+                    description = "kayttajaId is not a proper UUID",
+                    responseCode = "400",
+                ),
+                ApiResponse(
+                    description = "Hankekayttaja not found or not authorized",
+                    responseCode = "404",
+                ),
+            ]
+    )
+    @PreAuthorize(
+        "@featureService.isEnabled('USER_MANAGEMENT') && " +
+            "@hankeKayttajaAuthorizer.authorizeKayttajaId(#kayttajaId, 'DELETE_USER')"
+    )
+    fun checkForDelete(@PathVariable kayttajaId: UUID): DeleteInfo {
+        val kayttaja = hankeKayttajaService.getKayttaja(kayttajaId)
+        val hanke = hankeService.loadHankeById(kayttaja.hankeId)!!
```

Comment:
This will throw NPE and return 500 error but that is okay, because unless the database is totally bonkers there should always be a Hanke?

Reply:
I added an elvis operator to throw an exception.